src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,82 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { setCurrentTodo } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const [loaded, setLoaded] = useState(true);
```

Comment:
The `loaded` state is initially set to `true`, which means the `Loader` component will be shown initially. Consider setting it to `false` initially if you want the modal to open with the user data already loaded.